lib/puppet/util/windows/adsi.rb

```diff
@@ -236,6 +238,9 @@ def add_flag(flag_name, value)
 
     def password=(password)
       if !password.nil?
+        if password == ''
+          Puppet.warning("Blank (zero length) passwords are not recommended. Blank passwords may be allowed on your system based on policy, but Puppet cannot verify a blank password with the operating system. Because of this a password change event will be executed by Puppet on every run.")
```

Comment:
Style guide: "may" is an ambiguous word. In this case, we should replace it with "might."